src/nvim/eval.c

```diff
@@ -21686,6 +21788,71 @@ static inline bool eval_fname_sid(const char *const name)
   return *name == 's' || TOUPPER_ASC(name[2]) == 'I';
 }
 
+/// Return the declaration body of the function.
+///
+/// @param[in]  fp       Function pointer.
+/// @param[in]  replace  Add '!' to function head (i.e.: "function!").
+///
+/// @return allocated function declaration body string.
+char *get_func_body(ufunc_T *fp, bool replace)
```

Comment:
`:verbose func Foo` shows the body of a function. Impl is `ex_function` https://github.com/neovim/neovim/blob/b8fcf62da9261e47821c5e3d369dc4fd58d2b721/src/nvim/eval.c#L20997-L20999

I guess you already looked at it, because `get_func_body` seems to have many of special-cases covered.

Reply:
Yeah, I've copied a considerable amount from `list_func_head`. I'm thinking of refactoring it out of `list_func_head` into, say, `get_func_head`, and using it in both `list_func_head` and the context stuff. As for the body, I'm also thinking of keeping `get_func_body` limited to that and adding some flags to it (whether to include line numbers, indent lines, etc...) and then using it in both `ex_function` and the context stuff.

Reply:
Or maybe I should just capture the output of `:func {name}`, is it fine to use API handlers (e.g.: `nvim_command_output`) internally? I see it very rarely done (for example, [here](https://github.com/neovim/neovim/blob/master/src/nvim/eval.c#L8834-L8835))

Reply:
> Or maybe I should just capture the output of `:func {name}`, is it fine to use API handlers (e.g.: `nvim_command_output`) internally? I see it very rarely done (for example, [here](https://github.com/neovim/neovim/blob/master/src/nvim/eval.c#L8834-L8835))

Yes, totally fine. It's not common because it's not very "ergonomic", currently.

Reply:
> Do you want to attempt that in this PR?

Yeah, then we can merge this. I am worried about these two "indeterministic" fails in QuickBuild and Travis.